packages/pigeon/platform_tests/shared_test_plugin_code/lib/integration_tests.dart

```diff
@@ -105,37 +244,43 @@ void runPigeonIntegrationTests(TargetGenerator targetGenerator) {
     testWidgets('nested objects can be sent correctly', (WidgetTester _) async {
       final HostIntegrationCoreApi api = HostIntegrationCoreApi();
 
-      const String sentString = 'Some string';
-      final AllTypesWrapper sentObject =
-          AllTypesWrapper(values: AllTypes(aString: sentString));
+      final AllNullableTypesWrapper sentObject =
+          AllNullableTypesWrapper(values: genericAllNullableTypes);
 
-      final String? receivedString = await api.extractNestedString(sentObject);
-      expect(receivedString, sentString);
+      final String? receivedString =
+          await api.extractNestedNullableString(sentObject);
+      expect(receivedString, sentObject.values.aNullableString);
     });
 
     testWidgets('nested objects can be received correctly',
         (WidgetTester _) async {
       final HostIntegrationCoreApi api = HostIntegrationCoreApi();
 
       const String sentString = 'Some string';
-      final AllTypesWrapper receivedObject =
-          await api.createNestedString(sentString);
-      expect(receivedObject.values.aString, sentString);
+      final AllNullableTypesWrapper receivedObject =
+          await api.createNestedNullableString(sentString);
+      expect(receivedObject.values.aNullableString, sentString);
     });
 
     testWidgets(
         'Arguments of multiple types serialize and deserialize correctly',
         (WidgetTester _) async {
       final HostIntegrationCoreApi api = HostIntegrationCoreApi();
-      const String aString = 'this is aString';
-      const bool aBool = false;
-      const int anInt = 42;
-
-      final AllTypes echoObject =
-          await api.sendMultipleTypes(aBool, anInt, aString);
-      expect(echoObject.anInt, anInt);
-      expect(echoObject.aBool, aBool);
-      expect(echoObject.aString, aString);
+      const String aNullableString = 'this is a String';
+      const bool aNullableBool = false;
+      const int aNullableInt = 42;
+
+      final AllNullableTypes echoObject = await api.sendMultipleNullableTypes(
+          aNullableBool, aNullableInt, aNullableString);
+      expect(echoObject.aNullableInt, aNullableInt);
+      expect(echoObject.aNullableBool, aNullableBool);
+      expect(echoObject.aNullableString, aNullableString);
+
+      final AllNullableTypes echoNullFilledObject =
```

Comment:
Optional: For this and all of the following ones I would make the sending null version a separate test instead of testing two calls in the same unit test, so a test failure will correspond exactly to one call.

Reply:
I was torn on that. Will do though.